ee/server/service/software_installers.go

```diff
@@ -108,3 +109,83 @@ func (svc *Service) DeleteSoftwareInstaller(ctx context.Context, id uint) error
 
 	return nil
 }
+
+func (svc *Service) GetSoftwareInstallerMetadata(ctx context.Context, installerID uint) (*fleet.SoftwareInstaller, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.SoftwareInstaller{}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+
+	// get the installer's metadata
+	meta, err := svc.ds.GetSoftwareInstallerMetadata(ctx, installerID)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting software installer metadata")
+	}
+
+	// authorize with the software installer's team id
+	if err := svc.authz.Authorize(ctx, &fleet.SoftwareInstaller{TeamID: meta.TeamID}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+
+	return meta, nil
+}
+
+func (svc *Service) DownloadSoftwareInstaller(ctx context.Context, installerID uint) (*fleet.DownloadSoftwareInstallerPayload, error) {
+	meta, err := svc.GetSoftwareInstallerMetadata(ctx, installerID)
+	if err != nil {
+		return nil, err
+	}
+
+	return svc.getSoftwareInstallerBinary(ctx, meta.StorageID, meta.Name)
+}
+
+func (svc *Service) OrbitDownloadSoftwareInstaller(ctx context.Context, installerID uint) (*fleet.DownloadSoftwareInstallerPayload, error) {
+	// this is not a user-authenticated endpoint
+	svc.authz.SkipAuthorization(ctx)
+
+	// TODO: confirm error handling
```

Comment:
error handling looks pretty good, is there anything else you had in mind?

Reply:
Not specifically. More of a general sanity check on how/when to use the `fleet.OrbitError{}`, thoughts on the messages for the various error exits from this method.

Reply:
gotcha yeah, I couldn't find any special usage of `OrbitError`, I see the orbit endpoints use it sometimes. I think it's fine!